src/libcore/convert.rs

```diff
@@ -442,14 +470,3 @@ impl AsRef<str> for str {
         self
     }
 }
-
-// FromStr implies TryFrom<&str>
```

Comment:
Looks like this broke something in Travis:
```
[00:53:44] error[E0277]: the trait bound `char: core::convert::From<&str>` is not satisfied
[00:53:44]   --> /checkout/src/libcore/../libcore/tests/char.rs:35:16
[00:53:44]    |
[00:53:44] 35 |     assert_eq!(char::try_from("a").unwrap(), 'a');
[00:53:44]    |                ^^^^^^^^^^^^^^ the trait `core::convert::From<&str>` is not implemented for `char`
[00:53:44]    |
[00:53:44]    = help: the following implementations were found:
[00:53:44]              <char as core::convert::From<u8>>
[00:53:44]    = note: required because of the requirements on the impl of `core::convert::TryFrom<&str>` for `char`
```